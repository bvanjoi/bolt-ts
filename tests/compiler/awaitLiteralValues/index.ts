// From `github.com/microsoft/TypeScript/blob/v5.9.3/tests/cases/compiler/awaitLiteralValues.ts`, Apache-2.0 License

function awaitString() {
    await 'literal';
    //~^ ERROR: 'await' expressions are only allowed within async functions and at the top levels of modules.
}

function awaitNumber() {
    await 1;
    //~^ ERROR: 'await' expressions are only allowed within async functions and at the top levels of modules.
}

function awaitTrue() {
    await true;
    //~^ ERROR: 'await' expressions are only allowed within async functions and at the top levels of modules.
}

function awaitFalse() {
    await false;
    //~^ ERROR: 'await' expressions are only allowed within async functions and at the top levels of modules.
}

function awaitNull() {
    await null;
    //~^ ERROR: 'await' expressions are only allowed within async functions and at the top levels of modules.
}

function awaitUndefined() {
    await undefined;
    //~^ ERROR: 'await' expressions are only allowed within async functions and at the top levels of modules.
}
