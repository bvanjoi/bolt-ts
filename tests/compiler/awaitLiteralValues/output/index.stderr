  × 'await' expressions are only allowed within async functions and at the top
  │ levels of modules.
   ╭─[../../tests/compiler/awaitLiteralValues/index.ts:4:5]
 4 │ await
   · ─────
   ╰────

  × 'await' expressions are only allowed within async functions and at the top
  │ levels of modules.
   ╭─[../../tests/compiler/awaitLiteralValues/index.ts:9:5]
 9 │ await
   · ─────
   ╰────

  × 'await' expressions are only allowed within async functions and at the top
  │ levels of modules.
    ╭─[../../tests/compiler/awaitLiteralValues/index.ts:14:5]
 14 │ await
    · ─────
    ╰────

  × 'await' expressions are only allowed within async functions and at the top
  │ levels of modules.
    ╭─[../../tests/compiler/awaitLiteralValues/index.ts:19:5]
 19 │ await
    · ─────
    ╰────

  × 'await' expressions are only allowed within async functions and at the top
  │ levels of modules.
    ╭─[../../tests/compiler/awaitLiteralValues/index.ts:24:5]
 24 │ await
    · ─────
    ╰────

  × 'await' expressions are only allowed within async functions and at the top
  │ levels of modules.
    ╭─[../../tests/compiler/awaitLiteralValues/index.ts:29:5]
 29 │ await
    · ─────
    ╰────
